application-diagram-ui/src/main/resources/Diagram/DiagramEditSheet.xml

```diff
@@ -635,44 +635,36 @@ define('diagram-store', ['jquery', 'xwiki-meta', 'xwiki-utils', 'diagram-utils',
   };
 
   var getTagName = function(element) {
-    if (!element) { return ''; }
+    if (!element || isTextNode(element)) { return ''; }
     return element.tagName.toLowerCase();
   };
 
-  var isTextElement = function(element) {
-    if (!element) { return false; }
-    return element.nodeType == 3;
+  var isTextNode = function(element) {
+    return element &amp;&amp; element.nodeType === 3;
   };
 
   var isHeadingElement = function(element) {
-    if (!element) { return false; }
-    return !isTextElement(element) &amp;&amp; ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'].indexOf(getTagName(element)) != -1;
+    return  ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'].indexOf(getTagName(element)) != -1;
   };
 
   var isBlockElement = function(element) {
-    if (!element) { return false; }
-    return !isTextElement(element) &amp;&amp;
-      (['p', 'div'].indexOf(getTagName(element)) != -1 || isHeadingElement(element));
+    return ['p', 'div'].indexOf(getTagName(element)) != -1 || isHeadingElement(element);
   };
 
   var isListElement = function(element) {
-    if (!element) { return false; }
-    return !isTextElement(element) &amp;&amp; ['ul', 'ol'].indexOf(getTagName(element)) != -1;
+    return ['ul', 'ol'].indexOf(getTagName(element)) != -1;
   };
 
   var isInlineElement = function(element) {
-    if (!element) { return false; }
-    return !isTextElement(element) &amp;&amp; ['b', 'i', 'span'].indexOf(getTagName(element)) != -1;
+    return  ['b', 'i', 'span'].indexOf(getTagName(element)) != -1;
```

Comment:
```suggestion
    return  ['b', 'i', 'span'].indexOf(getTagName(element)) !== -1;
```
But I find it safer and more natural to check for ``>= 0`` (imagine there is a bug in the native API and it returns -2 instead of -1 :) )